plugins/woocommerce/client/legacy/js/frontend/a8c-address-autocomplete-service.js

```diff
@@ -0,0 +1,485 @@
+			return leading
+				? invokeFunc( time )
+				: new Promise( ( resolve ) => {
+						// Store the resolve function to be called when the function executes
+						lastArgs._resolve = resolve;
+				  } );
+		}
+
+		function remainingWait( time ) {
+			var timeSinceLastCall = time - lastCallTime,
+				timeSinceLastInvoke = time - lastInvokeTime,
+				timeWaiting = wait - timeSinceLastCall;
+
+			return maxing
+				? Math.min( timeWaiting, maxWait - timeSinceLastInvoke )
+				: timeWaiting;
+		}
+
+		function shouldInvoke( time ) {
+			var timeSinceLastCall = time - lastCallTime,
+				timeSinceLastInvoke = time - lastInvokeTime;
+
+			// Either this is the first call, activity has stopped and we're at the
+			// trailing edge, the system time has gone backwards and we're treating
+			// it as the trailing edge, or we've hit the `maxWait` limit.
+			return (
+				lastCallTime === undefined ||
+				timeSinceLastCall >= wait ||
+				timeSinceLastCall < 0 ||
+				( maxing && timeSinceLastInvoke >= maxWait )
+			);
+		}
+
+		function timerExpired() {
+			var time = Date.now();
+			if ( shouldInvoke( time ) ) {
+				return trailingEdge( time );
+			}
+			// Restart the timer.
+			timerId = setTimeout( timerExpired, remainingWait( time ) );
+		}
+
+		function trailingEdge( time ) {
+			timerId = undefined;
+
+			// Only invoke if we have `lastArgs` which means `func` has been
+			// debounced at least once.
+			if ( trailing && lastArgs ) {
+				return invokeFunc( time );
+			}
+			lastArgs = lastThis = undefined;
+			return result;
+		}
+
+		function cancel() {
+			if ( timerId !== undefined ) {
+				clearTimeout( timerId );
+			}
+			// Reject any pending promise
+			if ( lastArgs && lastArgs._resolve ) {
+				lastArgs._resolve( [] ); // Resolve with empty array for cancelled requests
+			}
+			lastInvokeTime = 0;
+			lastArgs = lastCallTime = lastThis = timerId = undefined;
+		}
+
+		function flush() {
+			return timerId === undefined ? result : trailingEdge( Date.now() );
+		}
+
+		function debounced() {
+			var time = Date.now(),
+				isInvoking = shouldInvoke( time );
+
+			lastArgs = arguments;
+			lastThis = this;
+			lastCallTime = time;
+
+			if ( isInvoking ) {
+				if ( timerId === undefined ) {
+					return leadingEdge( lastCallTime );
+				}
+				if ( maxing ) {
+					// Handle invocations in a tight loop.
+					clearTimeout( timerId );
+					timerId = setTimeout( timerExpired, wait );
+					return invokeFunc( lastCallTime );
+				}
+			}
+			if ( timerId === undefined ) {
+				timerId = setTimeout( timerExpired, wait );
+			}
+			// Return a promise that will resolve when the function is eventually called
+			return new Promise( ( resolve ) => {
+				// Store the resolve function to be called when the function executes
+				lastArgs._resolve = resolve;
+			} );
+		}
+		debounced.cancel = cancel;
+		debounced.flush = flush;
+		return debounced;
+	}
+
+	Object.entries( a8cAddressAutocompleteServiceKeys ).forEach(
```

Comment:
This script will be enqueued once for all providers, they all just into this global var and add their tokens.